### PR TITLE
[EDU-1627] - Spaces Avatar Stack example

### PR DESCRIPTION
**Pull request by GregHolmes**
This PR contains the two code samples for the new Examples section we are creating. It contains a Typescript and a React version for Avatar Stack with Spaces. Which will be rendered in the code sandbox live examples in the future, along with others.

The README within each directory /examples/spaces-avatar-stack/javascript and /examples/spaces-avatar-stack/react contains information on how to get these running.

[Jira epic ticket](https://ably.atlassian.net/browse/EDU-1541)
[Jira ticket](https://ably.atlassian.net/browse/EDU-1627).

### CODE REVIEW

File: examples/spaces-avatar-stack/javascript/src/script.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    });
+
+    avatar.addEventListener('mouseleave', () => {
+      popup.style.display = 'none';
+    });
+  }
+}
+
+function renderExceedingCounter(otherMembers: SpaceMember[]) {
+  console.log('here');
```
Comment (ttypic) [inline examples/spaces-avatar-stack/javascript/src/script.ts:137]: Looks something left after debugging

File: examples/spaces-avatar-stack/react/src/components/Avatar.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        style={{
+          backgroundColor: user.profileData.memberColor
+        }}
+        onMouseOver={() => setHover(true)}
+        onMouseLeave={() => setHover(false)}
+        id="avatar"
+      >
+        <p className={"textWhite nameOthers"}>{userInitials}</p>
+      </div>
+      {hover && self ? (
```
Comment (ttypic) [inline examples/spaces-avatar-stack/react/src/components/Avatar.tsx:68]: I would've done this without react, css only, this way we can delete all lines with hover. css will look something like:

```css
.avatar:not(:hover)  + .popup {
  display: none;
}
```

**Review by ttypic — COMMENTED** [review]
LGTM, left couple of small comments

File: examples/spaces-avatar-stack/react/src/components/Avatar.tsx (hunk 2 above)
Comment (GregHolmes) [reply examples/spaces-avatar-stack/react/src/components/Avatar.tsx]: Ohh! That is much simpler. Thanks. I've implemented it.

File: examples/spaces-avatar-stack/page.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
+# Display active users avatars in an application.
+
+Display online users through an avatar stack in an application.
```
Comment (m-hulbert) [inline examples/spaces-avatar-stack/page.md:3]:
```suggestion
Display the online status of users by displaying an avatar for each of them.
```

File: examples/spaces-avatar-stack/page.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+Display an the status of users within an application with the avatar stack to provide a visual rendering of which users are in the shared workspace.
```
Comment (m-hulbert) [inline examples/spaces-avatar-stack/page.md:5]:
```suggestion
Display the status of users within an application with an avatar stack to provide a visual rendering of which users are in the shared workspace.
```

File: examples/spaces-avatar-stack/page.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+Avatar stacks are utilized in applications such as online whiteboards, slide shows, spreadsheets and documents to let others know who is currently viewing the application or document.
```
Comment (m-hulbert) [inline examples/spaces-avatar-stack/page.md:7]:
```suggestion
Avatar stacks are utilized in applications such as online whiteboards, slide shows, spreadsheets and documents to let others know who is currently active in the application.
```

**Review by m-hulbert — CHANGES_REQUESTED** [review]
Some minor comments on the content and one small issue with the example: 👍 

I managed to get the same name and same avatar stack colours in JS, we should account for that if possible.

**Review by m-hulbert — APPROVED** [review]

**Pull request merged**